### PR TITLE
Release v1.0.0

### PR DESCRIPTION
**Pull request by IshaanDesai**
## Description

This is the first major release of the adapter. The release introduces some majors and breaking changes.

- The [code's repository](https://github.com/precice/dumux-adapter) moved to GitHub and is part of the preCICE project now. The CI has been ported and extended to use GitHub Actions accordingly.
- preCICE v2 is required.
- The code is licensed under GPLv3. This is the same license as used by [DuMuX](https://dumux.org).
- All monolithicaly coupled examples have been removed since they do not fit the purpose of the adapter and make maintenance more tedious.
- Initial support for exchanging vector quantities has been added.
- Ensures compatibility with new DuMuX versions including DuMuX 3.5 and the current DuMuX master branch.
- The test and CI have been enhanced to check the code and documentation style.
- Several Docker images for testing have been set up for testing. You can find the recipes in the `docker/` directory or on DockerHub as `precice/dumux-precice` or `dune-precice`.
- The paths and names of some headers and classes have changed. Now one has to include `dumux-precice/couplingadapter.hh` to use the adapter class `CouplingAdapter`. The adapter class resides in the namespace `Dumux::Precice`
- A dummy solver (`examples/dummysolver/`) has been added that shows how one can use the adapter.
- The documentation has been extended. There is a Doxygen documentation (`doc/doxygen`) of the API and the is some user documentation (`doc/user/`) based on mkdocs.

For further changes, fixes and additional details please consult the `CHANGELOG.md` in the repository.
## Checklist

- [x] I made sure that the source files are formatted properly.
- [x] I added my changes to the changelog (`CHANGELOG.md`)
- [x] I updated the documentation.

### CODE REVIEW

**Comment by ajaust** [comment]
This is the first major release of the adapter. The release introduces some majors and breaking changes.

- The [code's repository](https://github.com/precice/dumux-adapter) moved to GitHub and is part of the preCICE project now. The CI has been ported and extended to use GitHub Actions accordingly.
- preCICE v2 is required.
- The code is licensed under GPLv3. This is the same license as used by [DuMuX](https://dumux.org).
- All monolithicaly coupled examples have been removed since they do not fit the purpose of the adapter and make maintenance more tedious.
- Initial support for exchanging vector quantities has been added.
- Ensures compatibility with new DuMuX versions including DuMuX 3.5 and the current DuMuX master branch.
- The test and CI have been enhanced to check the code and documentation style.
- Several Docker images for testing have been set up for testing. You can find the recipes in the `docker/` directory or on DockerHub as `precice/dumux-precice` or `dune-precice`.
- The paths and names of some headers and classes have changed. Now one has to include `dumux-precice/couplingadapter.hh` to use the adapter class `CouplingAdapter`. The adapter class resides in the namespace `Dumux::Precice`
- A dummy solver (`examples/dummysolver/`) has been added that shows how one can use the adapter.
- The documentation has been extended. There is a Doxygen documentation (`doc/doxygen`) of the API and the is some user documentation (`doc/user/`) based on mkdocs.

For further changes, fixes and additional details please consult the `CHANGELOG.md` in the repository.

File: dune.module [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 #Name of the module
 Module: dumux-precice
-Version: develop
+Version: 1.0.0
 Maintainer: alexander.jaust@ipvs.uni-stuttgart.de
```
Comment (ajaust) [inline dune.module:8]: Do want to change the email address/the maintainer here?

**Review by ajaust — APPROVED** [review]
Looks overall good to me. There is just one small question regarding who to put as maintainer.

File: dune.module (hunk 1 above)
Comment (IshaanDesai) [reply dune.module]: Yes, I have changed it to my email.

Comment (ajaust) [reply dune.module]: Great. :+1:

**Pull request merged**